docs/search-v2-change-indices.md

```diff
@@ -0,0 +1,121 @@
+2. [Get the settings of the index in prod](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-settings.html)
+3. [Create a new index locally](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html),
+using parts of the above responses:
+
+    ```
+    PUT /my-index-000001
+    {
+      "mappings": {
+        < prod_index_name/_mapping["prod_index_name"]["mappings"] >
+      },
+      "settings: {
+        "analyzers": {
+          < prod_index_name/_settings["prod_index_name"]["settings"]["index"]["analysis"] >
+        }
+      }
+    }
+    ```
+
+4. Point your Document at this index and run `./manage.py es7_init --limit DocumentName`
+
+If this succeeds,
+you can be relatively confident that this is a config change which doesn't require creating a new index.
+
+## Changes appended to a current index
+
+Your approach here will completely depend on what kind of changes you're making on the index.
+
+For example,
+if you're adding a new field which hasn't existed before,
+you'll want to make sure any changes you make which read or query ElasticSearch don't assume that the field is immediately available.
+This is because,
+until you've fully reindexed the documents in this index,
+documents will exist without the field you're adding.
+
+So if you were adding a `searchme` field to a document which defaults to true,
+rather than adding a filter to only include documents where `searchme=True`,
+you would want your query to exclude documents where `searchme=False`,
+in order to avoid having a moment during your deploy where nothing appears in the query results.
+
+You could also use waffle flags in the same way as described below.
+
+```eval_rst
+.. Note::
+  If you're removing a field from a Document,
+  it would be preferable to leave it in the Document class,
+  setting its value on all documents to `None`,
+  with a note to remove it the next time a change requiring a new index is made.
+
+  This is to avoid the index on prod containing fields which don't exist on indices created afresh locally.
+```
+
+## Changes requiring a new index
+
+### Development process
+
+#### Config changes
+
+* Make any changes to Documents, analyzers or index configuration which you need to do as usual.
+* Any indices which are touched by these changes must have a new, separate, write index configured.
+
+For instance,
+if you make a change to the analyzers used across the wiki and question indices,
+named `sumo_wiki_document` and `sumo_question_1`,
+configure the Documents using those indices to now use `sumo_wiki_document_1` and `sumo_question_2`.
+
+#### Write operations
+
+These will be run against the new index immediately,
+as the new index will be initialized at deploy time by the `bin/run-post-deploy.sh` script.
+
+#### Read operations
+
+Place the changes behind a waffle switch,
```

Comment:
I would prefer not to use waffle and add another dependency in our indexing policies. This must be an autonomous process that is not dependent on any external library.

One approach that we could follow is to use two aliases. One for read and one for write. The read alias though can point to both the new and the old index. This will help as avoid stale data but will create duplicate results. For that case we could conditionally collapse on the ID when the aliases do not point to the exact same index. 

If this is to big of an overhead, we could either use two aliases, one for read and one for write and tolerate stale data while the indexing process is active or use a single one. In any case, this will give us an autonomous indexing process that has no external dependencies. I fear that waffle is over complicating things without any benefits, at least at a first glance. Please let me know if there is anything around waffle though though that can weight on our decision.